src/common/libfluxutil/method.c

```diff
@@ -125,9 +125,13 @@ void method_rusage_cb (flux_t *h,
                        void *arg)
 {
     struct rusage ru;
-
+#ifdef RUSAGE_THREAD
+    int who = RUSAGE_THREAD;
+#else
+    int who = RUSAGE_SELF;
+#endif
     if (flux_request_decode (msg, NULL, NULL) < 0
-        || getrusage (RUSAGE_THREAD, &ru) < 0)
+        || getrusage (who, &ru) < 0)
```

Comment:
OK, I just pushed a change that adds an optional `who` key to the RPC.  The default is to use RUSAGE_SELF but the other `who` values can be requested.  Then I changed `flux module stats --rusage` to work as follows
```
$ flux module stats --rusage=[self|children|thread]
```
so there is now an optional argument.  This is now documented it in flux-module(1).

On systems where RUSAGE_THREAD is unsupported, you'll get an error like this:
```
$ flux module stats --rusage=thread connector-local
flux-module: connector-local.rusage: thread is unsupported
```

Since this is little used and the documentation before this was ambiguous, I think this change is probably not disruptive to anything.

Reply:
LGTM, maybe wanna split off those 4 commits to a separate PR since it is more than just portability at this point.